File: src/main/java/com/team3/gudit/user/dto/UserMeResponseDto.java
package com.team3.gudit.user.dto;

import com.team3.gudit.user.domain.entity.Role;
import com.team3.gudit.user.domain.entity.User;

public record UserMeResponseDto(
        Long id,
        String nickname,
        String email,
        Role role
) {
    public static UserMeResponseDto from(User user) {
        return new UserMeResponseDto(
                user.getId(),
                user.getNickname(),
                user.getEmail(),
                user.getRole()
        );
    }
}

File: src/main/kotlin/com/team3/gudit/user/dto/UserMeResponseDto.kt
package com.team3.gudit.user.dto

import com.team3.gudit.user.domain.entity.Role
import com.team3.gudit.user.domain.entity.User

data class UserMeResponseDto(
    val id: Long?,
    val nickname: String?,
    val email: String?,
    val role: Role,
) {
    companion object {
        @JvmStatic
        fun from(user: User): UserMeResponseDto = UserMeResponseDto(
            id = user.id,
            nickname = user.nickname,
            email = user.email,
            role = user.role,
        )
    }
}
